File: src/基础算法和其他/模拟/PAT_Advanced_Level_Practice1117_EddingtonNumber/Main2.dart
import 'dart:io';

void main() {
  int n = int.parse(stdin.readLineSync()!);
  List<int> arr = stdin.readLineSync()!.split(' ').map(int.parse).toList();

  arr.sort();

  int ans = 1;
  for (int j = arr.length - 1; j >= 0; j--, ans++) {
    if (arr[j] <= ans) {
      print(ans - 1);
      break;
    }
  }
}

File: src/基础算法和其他/模拟/PAT_Advanced_Level_Practice1117_EddingtonNumber/Main2.go
package PAT_Advanced_Level_Practice1117_EddingtonNumber

import (
	"bufio"
	"fmt"
	"os"
	"sort"
	"strconv"
	"strings"
)

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Scan()
	n, _ := strconv.Atoi(scanner.Text())

	arr := make([]int, n)
	scanner.Scan()
	numbers := strings.Fields(scanner.Text())
	for i := 0; i < n; i++ {
		arr[i], _ = strconv.Atoi(numbers[i])
	}

	sort.Ints(arr)

	for j, ans := n-1, 1; j >= 0; j, ans = j-1, ans+1 {
		if arr[j] <= ans {
			fmt.Println(ans - 1)
			break
		}
	}
}